app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/xwatch/XWatchTaskerSpec.java

```diff
@@ -0,0 +1,41 @@
+package nodomain.freeyourgadget.gadgetbridge.devices.xwatch;
+
+import android.bluetooth.BluetoothGatt;
+import android.bluetooth.BluetoothGattCharacteristic;
+
+import java.util.Arrays;
+import java.util.List;
+
+import nodomain.freeyourgadget.gadgetbridge.model.DeviceType;
+import nodomain.freeyourgadget.gadgetbridge.tasker.event.TaskerEventType;
+import nodomain.freeyourgadget.gadgetbridge.tasker.spec.AbstractTaskerSpec;
+
+public class XWatchTaskerSpec extends AbstractTaskerSpec {
+
+    public XWatchTaskerSpec(DeviceType device) {
+        super(device);
+    }
+
+    @Override
+    public TaskerEventType getEventType(BluetoothGatt gatt, BluetoothGattCharacteristic characteristic) {
+        if (XWatchService.UUID_NOTIFY.equals(characteristic.getUuid())) {
+            byte[] data = characteristic.getValue();
+            if (data[0] == XWatchService.COMMAND_ACTIVITY_DATA) {
+                return TaskerEventType.DATA;
+            }
+            if (data[0] == XWatchService.COMMAND_ACTION_BUTTON) {
+                return TaskerEventType.BUTTON;
+            }
+            if (data[0] == XWatchService.COMMAND_CONNECTED) {
+                return TaskerEventType.CONNECTION;
```

Comment:
I think that such events would rather be handled generically via GBDevice#getState() changes (see GBDevice#sendDeviceUpdateEvent()).

Reply:
Can you give me more info on that.